### ISSUE TITLE
Generating the slider element programmatically

### ISSUE THREAD

**Issue by SterlingButters** [soundar24/roundSlider#102]
I having some problems doing the following:

```
  var slider = document.createElement('div');
  slider.id = 'slider';
  parentElement.appendChild(slider);

  $("#slider").roundSlider({
    sliderType: "range",
    circleShape: "pie",
    ...
  });
```

The roundSlider isn't created immediately since jQuery can't select the new div. Are there any workarounds?

**Comment by soundar24** [comment]
@SterlingButters the only possibility for the issue is, before you are initializing the slider, the slider element won't appended in the document body. I mean you append inside the `parentElement` but the parent element may still not appended in the body.

Check the below working demo, if you append the  the element already:
https://jsfiddle.net/soundar24/09ejun8b/2/

Suppose if you want to create the slider on the fly, then only you want to append into the page then on the slider initialization instead of the 'id' selector directly give the created element object. then it works, check the below demo:
https://jsfiddle.net/soundar24/09ejun8b/3/

**Comment by SterlingButters** [comment]
@soundar24 Please the problem in my example (sorry it is so hefty): https://codepen.io/sterlingbutters/pen/ZEbzKgz?editors=1111

Hit the "`+`" icon for "Room/Zone Panel" and you will see my issue. 

As you can see in Js lines 104-132 & 152, the slider is appended to the body but only on the 2nd press of the "`+`" icon. Any ideas?

**Comment by soundar24** [comment]
@SterlingButters as I mentioned in the previous comment, you can't use jQuery selector until the element was append into the body. I have corrected the code, you can check the below working demo:

https://codepen.io/soundar24/pen/YzyyNBp

Also at this case the tooltip position may misalign, so I just refreshed at line 153. And I will provide a refresh method for this in the next release.

Please check and let me know if you are facing any other issues.

**Comment by SterlingButters** [comment]
@soundar24 Ahh yes I see now, thank you so much!

**Comment by soundar24** [comment]
@SterlingButters  A new update for you.. the tooltip related problems are completely resolved on **[roundSlider v1.5.1](https://github.com/soundar24/roundSlider/releases/tag/v1.5.1)**, in addition I have provided the "**[refreshTooltip](https://roundsliderui.com/document.html#refreshTooltip)**" method with this version.

Hope this helps you, thanks...
